src/API/sim_utils.jl

```diff
@@ -1,11 +1,18 @@
-function load_netlist(netlist::Array{String})
+function load_netlist(netlist::AbstractArray{T}) where T <: AbstractString
     append!(netlist, ["C_NULL"])
     t = netlist |> ngSpice_Circ
 end
 
-function load_netlist(netpath::String)
-    netlist = readlines(netpath)
-    load_netlist(netlist)
+function load_netlist(net::AbstractString)
+    isfile(net) && (netlist = readlines(net);
+        load_netlist(netlist);
+        return)
+    # The strings in Julia always use \n instead of \r \r\n
+    # (even if the editor does).
+    # But if the circuit-string has \r in it,
+    # uncomment the following line
+    # occursin("\r", net) && replace(net, "\r"=>"")
+    load_netlist(split(net, "\n"))
```

Comment:
If the user passes in a bad filename then this will try to read it in as the contents of the netlist and give a weird error.  I'd like to have a separate function for reading a file vs a string that is a netlist in order to avoid this.  Ngspice uses `source` to read in a file so can we call a function by that name to read in file?

Reply:
Yeah, as we already have `source` function, we can restrict `load_netlist` to take only the netlist (in an array of strings or a string formats) and never the file name.